infocmdb/ci_type.go

```diff
@@ -93,3 +94,52 @@ func (c *Client) GetCiTypeName(ciId int) (ciTypeName string, err error) {
 
 	return
 }
+
+type respSetTypeOfCi struct {
+	Success bool   `json:"success"`
+	Message string `json:"message"`
+}
+
+func (c *Client) SetTypeOfCi(ciId int, ciType string) (err error) {
+
+	if err = c.v2.Login(); err != nil {
+		return
+	}
+
+	currentCiTYpe, err := c.GetCiTypeName(ciId)
+	if err != nil {
+		return err
+	}
+
+	if ciType == currentCiTYpe {
+		return errors.New("the requested ci type is already set: " + currentCiTYpe)
+	}
+
+	ciTypeId, err := c.GetCiTypeIdByCiTypeName(ciType)
+	if err != nil {
+		return err
+	}
+
+	ciTypeIdString := strconv.Itoa(ciTypeId)
+	ciIdString := strconv.Itoa(ciId)
+
+	params := map[string]string{
+		"argv1": ciIdString,
+		"argv2": ciTypeIdString,
+		"argv3": "0",
+	}
+
+	response := respSetTypeOfCi{}
+	err = c.v2.Query("int_setCiTypeOfCi", &response, params)
+	if err != nil {
+		err = utilError.FunctionError(err.Error())
+		log.Error("Error: ", err)
+		return
+	}
+
+	if response.Success != true {
+		return errors.New("couldn't change ci type to: " + ciType + " for ciid: " + ciIdString + " ,error: " + response.Message)
```

Comment:
The formatting of the error message looks odd to me (missing space before error), 
that might be personal preference though.

`couldn't change ci type to: some_ci_type for ciid: 12345 ,error: access denied`

Reply:
How would you format it? Im always open to suggestions.